src/Sulu/Bundle/AdminBundle/Resources/js/components/Modal/Modal.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import React from 'react';
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+import Portal from 'react-portal';
+import classnames from 'classnames';
+import {afterElementsRendered} from '../../services/DOM';
+import Backdrop from '../Backdrop';
+import ModalBox from './ModalBox';
+import type {ModalProps} from './types';
+import modalStyles from './modal.scss';
+
+type Props = ModalProps & {
+    isOpen: boolean,
+    onRequestClose: () => void,
+};
+
+const ESC_KEY = 27;
```

Comment:
If I remember correctly @chirimoya found a react library handling that using decorators.